python/ray/services.py

```diff
@@ -691,7 +719,7 @@ def _start_redis_instance(node_ip_address="127.0.0.1",
             break
         port = new_port()
         counter += 1
-    if counter == num_retries:
+    else:
```

Comment:
Again, not super sure about this pattern. It is a bit cleaner, but can be really confusing.